ibis/common/pretty.py

```diff
@@ -0,0 +1,152 @@
+from __future__ import annotations
+
+import functools
+from typing import IO, Any
+
+import sqlalchemy as sa
+import sqlalchemy.dialects.mysql
+import sqlalchemy.dialects.postgresql
+from multipledispatch import Dispatcher
+from sqlalchemy.engine import Dialect
+from sqlalchemy.sql import ClauseElement
+
+import ibis
+import ibis.common.exceptions as com
+import ibis.expr.types as ir
+
+_get_compiled_string = Dispatcher("_get_compiled_string")
+
+
+@_get_compiled_string.register(str)
+def _get_compiled_string_str(s: str) -> str:
+    return s
+
+
+@_get_compiled_string.register(ir.Expr)
+def _get_compiled_string_expr(expr: ir.Expr) -> str:
+    backend = expr._find_backend()
+    return _get_compiled_string(
+        expr.compile(),
+        getattr(getattr(backend, "con", None), "dialect", None),
+    )
+
+
+@_get_compiled_string.register(str, type(None))
+def _get_compiled_string_default(s: str, _: None) -> str:
+    return s
+
+
+@_get_compiled_string.register(ClauseElement)
+def _get_compiled_string_sa(expr: ClauseElement) -> str:
+    # Use the postgresql dialect if `bind` is `None`
+    return _get_compiled_string(
+        expr,
+        getattr(expr.bind, "dialect", sa.dialects.postgresql.dialect()),
+    )
+
+
+@_get_compiled_string.register(ClauseElement, Dialect)
+def _get_compiled_string_sa_dialect(
+    expr: ClauseElement, dialect: Dialect
+) -> str:
+    return str(expr.compile(dialect=dialect))
+
+
+@_get_compiled_string.register(
+    ClauseElement,
+    (sa.dialects.mysql.dialect, sa.dialects.postgresql.dialect),
+)
+def _get_compiled_string_mysql_pg(expr: ClauseElement, _: Dialect) -> str:
+    return str(expr.compile(compile_kwargs=dict(literal_binds=True)))
+
+
+@functools.singledispatch
+def _get_backend_name(expr: ir.Expr | ClauseElement) -> str:
```

Comment:
The backends should already define their name via `backend.name` class field.

Reply:
Sure, but this attempts to find the name for both ibis expression **and** sqlalchemy expressions.